cpp/src/arrow/compute/exec/options.h

```diff
@@ -270,5 +270,23 @@ class ARROW_EXPORT SelectKSinkNodeOptions : public SinkNodeOptions {
   SelectKOptions select_k_options;
 };
 
+/// \brief Adapt an Table as a sink node
+///
+/// plan->exec_context()->executor() will be used to parallelize pushing to
+/// outputs, if provided.
+class ARROW_EXPORT TableSinkNodeOptions : public ExecNodeOptions {
+ public:
+  TableSinkNodeOptions(std::shared_ptr<Table>* output_table,
+                       std::shared_ptr<Schema> output_schema,
+                       Future<> finish = Future<>::Make())
```

Comment:
Hmm, I realize the tests do this, but there's no reason to let the user provide `Future<> finish` right? The consumer implementation can manage its own Future instance and mark it finished itself. (Why make the user manually mark the future finished? How is the user even supposed to know when the node finishes?) I think the tests just have this parameter to test things.

Reply:
This should be removed.

Reply:
I missed this, now updated.